CMakeLists.txt

```diff
@@ -0,0 +1,226 @@
+cmake_minimum_required (VERSION 2.8.11)
+project (libpd C)
+
+option(pd_utils "Compile utilities" ON)
+option(pd_extra "Compile extras" ON)
+option(pd_multi "Compile with multiple instance support" OFF)
+option(pd_locale "Set the LC_NUMERIC number format to the default C locale" ON)
+option(pd_build_c_examples "Builds C API example programs" OFF)
+
+if(WIN32)
+    set(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -DWINVER=0x502 -DWIN32 -D_WIN32")
+endif()
+if (MSVC)
+    set(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -DHAVE_STRUCT_TIMESPEC")
+    add_definitions("/D_CRT_SECURE_NO_WARNINGS /wd4091 /wd4996")
+    if(${CMAKE_SIZEOF_VOID_P} EQUAL 8)
+        set(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -DPD_LONGINTTYPE=\"long long\"")
+    endif()
+else()
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-int-to-pointer-cast -Wno-pointer-to-int-cast")
+    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS} -ffast-math -funroll-loops -fomit-frame-pointer -O3")
+    set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS} -g -O0")
+endif()
+add_definitions(-DPD=1 -DUSEAPI_DUMMY=1)
+
+if(NOT DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
+    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.7)
+endif()
+if(NOT DEFINED CMAKE_OSX_ARCHITECTURES)
+    set(CMAKE_OSX_ARCHITECTURES "i386;x86_64")
+endif()
+set(CMAKE_MACOSX_RPATH ON)
+
+foreach (_BUILD_TYPE RELEASE DEBUG)
+    set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${_BUILD_TYPE} ${PROJECT_SOURCE_DIR}/libs)
+    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_${_BUILD_TYPE} ${PROJECT_SOURCE_DIR}/libs)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_${_BUILD_TYPE} ${PROJECT_SOURCE_DIR}/libs)
+endforeach()
+
+set(PD_SOURCES
```

Comment:
It would have been better to keep the header files I think

Reply:
Header files in the sources is an [antipattern](http://voices.canonical.com/jussi.pakkanen/2013/03/26/a-list-of-common-cmake-antipatterns/)

Reply:
From your reference:
> The reason people put that in is probably because they think it is required to make CMake rebuild the target in case the header is changed.

No, it's because you can see and select them in your project like XCode. It's not a problem to add the header files in the sources with CMake.

Reply:
I see, I could add them again for your convenience.